bridges/ARDMediathekBridge.php

```diff
@@ -4,32 +4,87 @@ class ARDMediathekBridge extends BridgeAbstract {
 	const URI = 'https://www.ardmediathek.de';
 	const DESCRIPTION = 'Feed of any series in the ARD-Mediathek, specified by its path';
 	const MAINTAINER = 'yue-dongchen';
+	/*
+	 * Number of Items to be requested from ARDmediathek API
+	 * 12 has been observed on the wild
+	 * 29 is the highest successfully tested value
+	 * More Items could be fetched via pagination
+	 * The JSON-field pagination holds more information on that
+	 * @const PAGESIZE number of requested items
+	 */
+	const PAGESIZE = 29;
+	/*
+	 * The URL Prefix of the (Webapp-)API
+	 * @const APIENDPOINT https-URL of the used endpoint
+	 */
+	const APIENDPOINT = 'https://api.ardmediathek.de/page-gateway/widgets/ard/asset/';
+	/*
+	 * The URL prefix of the video link
+	 * URLs from the webapp include a slug containing titles of show, episode, and tv station.
+	 * It seems to work without that.
+	 * @const VIDEOLINKPREFIX https-URL prefix of video links
+	 */
+	const VIDEOLINKPREFIX = 'https://www.ardmediathek.de/video/';
+	/*
+	 * The requested width of the preview image
+	 * 432 has been observed on the wild
+	 * The webapp seems to also compute and add the height value
+	 * It seems to works without that.
+	 * @const IMAGEWIDTH width in px of the preview image
+	 */
+	const IMAGEWIDTH = 432;
+	/*
+	 * Placeholder that will be replace by IMAGEWIDTH in the preview image URL
+	 * @const IMAGEWIDTHPLACEHOLDER
+	 */
+	const IMAGEWIDTHPLACEHOLDER = '{width}';
 
 	const PARAMETERS = array(
 		array(
 			'path' => array(
-				'name' => 'Path',
+				'name' => 'Show Link or ID',
 				'required' => true,
-				'title' => 'Enter without trailing slash',
-				'defaultValue' => '45-min/Y3JpZDovL25kci5kZS8xMzkx'
+				'title' => 'Link to the show page or just its alphanumeric suffix',
+				'defaultValue' => 'https://www.ardmediathek.de/sendung/45-min/Y3JpZDovL25kci5kZS8xMzkx/'
 			)
 		)
 	);
 
 	public function collectData() {
 		date_default_timezone_set('Europe/Berlin');
 
-		$url = 'https://www.ardmediathek.de/sendung/' . $this->getInput('path') . '/';
-		$html = getSimpleHTMLDOM($url);
-		$html = defaultLinkTo($html, $url);
+		$pathComponents = explode('/', $this->getInput('path'));
+		if (empty($pathComponents)) {
+			$pathComponents = explode('/', SELF::PARAMETERS[0]['defaultValue']);
+		}
```

Comment:
`path` is required parameter. Why do we need to handle case, where it is empty?

Reply:
Try for example ArteBridge with these parameters `?action=display&bridge=Arte7&context=Collection+(Français)&colfr=&format=Html`. Although the parameter is required, A zero-length string can still be called in a deep-link. I found it a good practice not to access an array at keys it wouldn't have but to safely fall back to the default value.

Reply:
I see. 
In this case I suggest to raise exception using returnClientError.

Reply:
Wonderful. I'm doing that now.